### PR TITLE
travis config with a jasmine test for socks.ts

### PR DESCRIPTION
**Pull request by trevj**
This adds a config for Travis and a near-trivial test for socks.ts, which I picked because it has a few functions without any dependencies (no need for the DOM or to load Freedom, etc.). It's just a start to get us setup with a continuous build!

Caveats:
- no Selenium tests yet (that's a whole different kettle of balls of wax and I don't want it to block us adding more unit tests)
- this doesn't test any Freedom-dependant code...hopefully that won't be hard
- we'll probably want to write tests in TypeScript at some point...I don't think anything here precludes that

There are several commits in this branch so I'll be sure to rebase rather than merging to keep the history clean :-)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,4 +34,4 @@ For example:
 
 `curl -x socks5h://localhost:9999 www.google.com`
 
-There will be actual tests soon!
+There will be more tests soon!
```
Comment (keroserene) [inline README.md:37]: yay!

File: spec/socks-to-rtc/socks_spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ipv4Request = new Uint8Array([
+      Socks.VERSION5,
+      Socks.REQUEST_CMD.CONNECT,
+      0, // reserved
+      Socks.ATYP.IP_V4,
+      192, 168, 1, 1, // IP: 192.168.1.1
+      1200 >> 8, 1200 & 0xFF]); // port: 1200
+  });
+
+  it("reject wrongly sized requests", function() {
```
Comment (keroserene) [inline spec/socks-to-rtc/socks_spec.js]: nit: We should prefer single quotes over double quotes in most cases

**Comment by keroserene** [comment]
Really cool - just that one nit, and LGTM / feel free to merge!

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+ - "0.11"
```
Comment (willscott) [inline .travis.yml:3]: Probably worth testing the current stable "0.10"  as well. maybe try previous stable "0.8" version as well.  Having multiple runs per commit helps detect nondeterminism / flakyness.

File: src/socks-to-rtc/socks.ts [hunk 4]
```diff
@@ -88,10 +90,14 @@ module Socks {
    * Parse byte array into a SOCKS request object.
    */
   export function interpretSocksRequest(byteArray:Uint8Array) {
+    var result:any = {};
```
Comment (iislucas) [inline src/socks-to-rtc/socks.ts:93]: add TODO: for specifying the result's real type.

File: .travis.yml (hunk 3 above)
Comment (trevj) [inline .travis.yml:3]: Done!

**Pull request merged**